catchpoint-mixin/panels.libsonnet

```diff
@@ -0,0 +1,418 @@
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true')
+        + g.panel.timeSeries.standardOptions.withUnit('ms'),
+
+      topAvgConnectionSetupTimeNodeName:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Top average connection setup time by nodes',
+          targets=[t.topAvgConnectionSetupTimeNodeName],
+          description='The top average connection setup time among all node names over the specified interval.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true')
+        + g.panel.timeSeries.standardOptions.withUnit('ms'),
+
+      topAvgContentLoadingTimeTestName:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Top average content loading time',
+          targets=[t.topAvgContentLoadingTimeTestName],
+          description='The top average content loading time among all test names over the specified interval.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true')
+        + g.panel.timeSeries.standardOptions.withUnit('ms'),
+
+      topAvgContentLoadingTimeNodeName:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Top average content loading time by nodes',
+          targets=[t.topAvgContentLoadingTimeNodeName],
+          description='The top average content loading time among all node names over the specified interval.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true')
+        + g.panel.timeSeries.standardOptions.withUnit('ms'),
+
+      topAvgRedirectsTestName:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Top average redirects by tests',
+          targets=[t.topAvgRedirectsTestName],
+          description='The top average number of redirects among all test names over the specified interval.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true'),
+
+      topAvgRedirectsNodeName:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Top average redirects by nodes',
+          targets=[t.topAvgRedirectsNodeName],
+          description='The top average number of redirects among all node names over the specified interval.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true'),
+
+      alertsPanel:
+        alertList.new('Catchpoint alerts')
+        + alertList.options.UnifiedAlertListOptions.withAlertInstanceLabelFilter(this.grafana.variables.queriesGroupSelectorAdvanced),
+
+      topErrorsByTestName:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Top errors by tests',
+          targets=[t.topErrorsByTestName],
+          description='The top number of errors encountered among all test names over the specified interval.'
+        )
+        + g.panel.timeSeries.fieldConfig.defaults.custom.withSpanNulls('true'),
```

Comment:
I think this panel would benefit from having a right-aligned table legend with values so users don't have to hover to find theirs.
![image](https://github.com/grafana/jsonnet-libs/assets/1282515/d721d46e-ceaf-49a2-9702-83c698ef7057)

I would then make it full-width as well to make space for it

Reply:
I'll make it 2/3 of the width and reduce the alerts to 1/3. I like the idea of having the table on the right.

Reply:
Yeah I like this, that's a lot cleaner :smile: